src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtDecoder.cs

```diff
@@ -0,0 +1,293 @@
+            }
+
+            if (decodingErrorCallBack is null)
+            {
+                throw new ArgumentNullException(nameof(decodingErrorCallBack));
+            }
+
+            if (string.IsNullOrWhiteSpace(tokenParameters.Token))
+            {
+                throw new ArgumentNullException(nameof(tokenParameters.Token));
+            }
+
+            _decodingErrorCallBack = decodingErrorCallBack;
+
+            var tokenResult = new TokenResult();
+
+            try
+            {
+                IdentityModelEventSource.ShowPII = true;
+                var handler = new JwtSecurityTokenHandler();
+                JwtSecurityToken jwtSecurityToken = handler.ReadJwtToken(tokenParameters.Token);
+                tokenResult.Header = JsonHelper.Format(jwtSecurityToken.Header.SerializeToJson(), Indentation.TwoSpaces, false);
+                tokenResult.Payload = JsonHelper.Format(jwtSecurityToken.Payload.SerializeToJson(), Indentation.TwoSpaces, false);
+                tokenResult.TokenAlgorithm = tokenParameters.TokenAlgorithm;
+
+                if (decodeParameters.ValidateSignature)
+                {
+                    var signatureValid = ValidateTokenSignature(handler, decodeParameters, tokenParameters, tokenResult);
+                    if (!signatureValid)
+                    {
+                        return null;
+                    }
+                }
+            }
+            catch (Exception exception)
+            {
+                RaiseError(exception.Message);
+                return null;
+            }
+
+            return tokenResult;
+        }
+
+        private bool ValidateTokenSignature(
+            JwtSecurityTokenHandler handler,
+            DecoderParameters decodeParameters,
+            TokenParameters tokenParameters,
+            TokenResult tokenResult)
+        {
+            SigningCredentials? signingCredentials = GetValidationCredentials(tokenParameters);
+            var validationParameters = new TokenValidationParameters
+            {
+                ValidateIssuerSigningKey = true,
+                IssuerSigningKey = signingCredentials.Key,
+                TryAllIssuerSigningKeys = true,
+                ValidateActor = decodeParameters.ValidateActor,
+                ValidateLifetime = decodeParameters.ValidateLifetime,
+                ValidateIssuer = decodeParameters.ValidateIssuer,
+                ValidateAudience = decodeParameters.ValidateAudience
+            };
+
+            if (decodeParameters.ValidateIssuer)
+            {
+                if (!tokenParameters.ValidIssuers.Any())
+                {
+                    RaiseError(_localizedStrings.ValidIssuersError);
+                    return false;
+                }
+                validationParameters.ValidIssuers = tokenParameters.ValidIssuers;
+            }
+
+            if (decodeParameters.ValidateAudience)
+            {
+                if (!tokenParameters.ValidAudiences.Any())
```

Comment:
When possible, avoid using LINQ. It's cheaper to use `Count` property here.
Same thing on line 108

Reply:
Sorry, bad habit from work they don't like Count and prefer Any...